src/main/java/net/minecraftforge/common/config/Configuration.java

```diff
@@ -80,9 +89,16 @@
 
     public Configuration(){}
 
+    @Deprecated
+    public Configuration(CharSource staticSource, CharSink staticSink){
```

Comment:
I partially agre with you: concerning file part. It's kept because there's `PARENT` config file, and i'm unsure if i should remove it (if i do, i'll also remove all `file` references except `getFile`).
I should probably edit it in PR description.